addons/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/state/PowermaxPanelSettings.java

```diff
@@ -0,0 +1,968 @@
+            }
+            if (rawSettings[i] == null) {
+                rawSettings[i] = new Byte[0x100];
+                for (int j = 0; j < 0x100; j++) {
+                    rawSettings[i][j] = null;
+                }
+            }
+            for (int j = start; j <= end; j++) {
+                rawSettings[i][j] = data[index++];
+            }
+        }
+    }
+
+    private byte[] readSettings(PowermaxSendType msgType, int start, int end) {
+        byte[] message = msgType.getMessage();
+        int page = message[2] & 0x000000FF;
+        int index = message[1] & 0x000000FF;
+        return readSettings(page, index + start, index + end);
+    }
+
+    private byte[] readSettings(int page, int start, int end) {
+        int pageMin = page + start / 0x100;
+        int indexPageMin = start % 0x100;
+        int pageMax = page + end / 0x100;
+        int indexPageMax = end % 0x100;
+        int index = 0;
+        boolean missingData = false;
+        for (int i = pageMin; i <= pageMax; i++) {
+            int start2 = 0;
+            int end2 = 0xFF;
+            if (i == pageMin) {
+                start2 = indexPageMin;
+            }
+            if (i == pageMax) {
+                end2 = indexPageMax;
+            }
+            index += end2 - start2 + 1;
+            for (int j = start2; j <= end2; j++) {
+                if ((rawSettings[i] == null) || (rawSettings[i][j] == null)) {
+                    missingData = true;
+                    break;
+                }
+            }
+            if (missingData) {
+                break;
+            }
+        }
+        if (missingData) {
+            logger.debug("readSettings({}, {}, {}): missing data", page, start, end);
+            return null;
+        }
+        byte[] result = new byte[index];
+        index = 0;
+        for (int i = pageMin; i <= pageMax; i++) {
+            int start2 = 0;
+            int end2 = 0xFF;
+            if (i == pageMin) {
+                start2 = indexPageMin;
+            }
+            if (i == pageMax) {
+                end2 = indexPageMax;
+            }
+            for (int j = start2; j <= end2; j++) {
+                result[index++] = rawSettings[i][j];
+            }
+        }
+        return result;
+    }
+
+    private String readSettingsAsString(PowermaxSendType msgType, int start, int end) {
+        byte[] message = msgType.getMessage();
+        int page = message[2] & 0x000000FF;
+        int index = message[1] & 0x000000FF;
+        String result = null;
+        byte[] data = readSettings(page, index + start, index + end);
+        if ((data != null) && ((data[0] & 0x000000FF) != 0x000000FF)) {
+            result = "";
+            for (int i = 0; i < data.length; i++) {
+                boolean endStr = false;
+                switch (data[i] & 0x000000FF) {
+                    case 0:
+                        endStr = true;
+                        break;
+                    case 1:
```

Comment:
Are these the only special chars? Interesting...

Reply:
Maybe not but they are the only ones I was able to test.